r/src/array_from_vector.cpp

```diff
@@ -159,6 +159,9 @@ struct VectorToArrayConverter {
       if (s == NA_STRING) {
         RETURN_NOT_OK(binary_builder->AppendNull());
         continue;
+      } else {
+        // Make sure we're ingesting UTF-8
+        s = Rf_mkCharCE(Rf_translateCharUTF8(s), CE_UTF8);
```

Comment:
This `Rf_mkCharCE(Rf_translateCharUTF8(s), CE_UTF8)` is dropped in several places; should we factor this out to a macro or something?

Reply:
yeah maybe some sort of `Rf_mkCharUtf8()` or `Rf_mkUtf8()`